### ISSUE TITLE
add a local git repo for e2e test

### ISSUE THREAD

**Issue by par97** [RamenDR/ramen#1459]
Background: when we deploy argocd based application and appset, we need specify a git repo address for the application to deploy. If we have a local git server that would help reduce network risk.

Nir already has this for set up such a git server:
https://github.com/RamenDR/ramen/tree/main/test/gitlap

After the local git server is ready, we can access such a repo within the cluster like:
https://github.com/RamenDR/ramen/tree/main/test/gitlap#testing-the-server

If we clone the remote argocd example repo and push it to the local git repo, we should be able to deploy the argocd app using the local git repo instead of github using:

--repo=http://host.minikube.internal/argocd-example-apps.git

**Comment by ShyamsundarR** [comment]
This maybe required for other tests like expanding a PVC, or expanding a StatefulSet, when the source of the PVC/STS is from git.

I would set its priority lower than other e2e tasks though at present, and come back to this as we add further test cases.

**Comment by par97** [comment]
yes, this is Nir's suggestion that we could set up such a service later. So this issue is opened for tracking purpose.
